### PR TITLE
feat: Resources query planner

### PR DESCRIPTION
**Pull request by dbuduev**
#### Description

Creates endpoint to return a pre-evaluated query plan for the resource on behalf of the given principal according to relevant policies.

#### Checklist 

- [x] The PR title has the correct prefix 
- [x] PR is linked to the corresponding issue
- [x] All commits are signed-off (`git commit -s ...`) to provide the [DCO](https://developercertificate.org/)

TODO:
- [x] Tracing

### CODE REVIEW

**Review by charithe — COMMENTED** [review]
Brilliant job! Very clean implementation too. Well done 👍🏽 

I left some comments mainly as reminders to myself when we catch-up.

File: api/public/cerbos/request/v1/request.proto [hunk 1]
```diff
@@ -19,6 +19,55 @@ import "google/api/field_behavior.proto";
...
+    string policy_version = 5 [
+        (validate.rules).string.pattern = "^[[:word:]]*$",
+        (google.api.field_behavior) = OPTIONAL,
+        (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+            description: "The policy version to use to evaluate this request. If not specified, will default to the server-configured default version.",
+            pattern: "^[[:word:]]*$"
+            example: "\"default\""
+        }
+    ];
+
```
[the remark this answers is not in the document]
Comment (dbuduev) [reply api/public/cerbos/request/v1/request.proto:67]: It makes total sense to me! I was thinking about this, but I'd prefer to implement it later. Hopefully, we can change the API in a backward compatible way.

File: api/public/cerbos/svc/v1/svc.proto [hunk 2]
```diff
@@ -73,6 +73,17 @@ service CerbosService {
             description: "Get information about the server e.g. server version"
         };
     }
+
+    rpc ResourcesQueryPlan(cerbos.request.v1.ResourcesQueryPlanRequest) returns (cerbos.response.v1.ResourcesQueryPlanResponse) {
+        option (google.api.http) = {
+            post: "/api/plan/resources",
```
Comment (charithe) [inline api/public/cerbos/svc/v1/svc.proto:79]: As discussed, let's put this under `/api/x/plan/resources` for now to indicate that it's experimental and there are no backward compatibility guarantees.

**Review by charithe — APPROVED** [review]
LGTM. Nice one 🥇

**Pull request merged**